java-checks/src/test/files/checks/AssertionsCompletenessCheck.java

```diff
@@ -86,13 +89,42 @@ public void assertj() {
   }
 
   @Test
-  public void testTruthFramework() {
-    boolean b = true;
-    Truth.assertThat(b).isTrue();
-    String s = "Hello Truth Framework World!";
-    Truth.assertThat(s).contains("Hello");
-    Truth.assertThat(b); // Noncompliant
-    Truth.assertWithMessage("Invalid option").that(b).isFalse();
-    Truth.assertWithMessage("Invalid option").that(b); // Noncompliant
+  public void assertj_soft_assertions_without_assertAll() {
+    org.assertj.core.api.SoftAssertions softly = new org.assertj.core.api.SoftAssertions();
+    softly.assertThat(5).isLessThan(3);
+    softly.assertThat(1).isGreaterThan(2);
+  } // Noncompliant {{Add a call to 'assertAll' after all 'assertThat'.}}
+
+  @Test
+  public void assertj_soft_assertions_without_assertThat() {
+    org.assertj.core.api.SoftAssertions softly = new org.assertj.core.api.SoftAssertions();
+    softly.assertAll(); // Noncompliant {{Add one or more 'assertThat' before 'assertAll'.}}
   }
+
+  @Test
+  public void assertj_soft_assertions_try_with_resource() {
+    try(org.assertj.core.api.AutoCloseableSoftAssertions softly = new org.assertj.core.api.AutoCloseableSoftAssertions()) {
+      softly.assertThat(1).isLessThan(2);
+    } // Compliant, no need to call "assertAll()", it will be called by AutoCloseableSoftAssertions
+  }
+
+  @Test
+  public void assertj_soft_assertions_try_with_resource_without_assertThat() {
+    try(org.assertj.core.api.AutoCloseableSoftAssertions softly = new org.assertj.core.api.AutoCloseableSoftAssertions()) {
+    } // Noncompliant {{Add one or more 'assertThat' before 'assertAll'.}}
+  }
+
+  @Test
+  public void assertj_soft_assertions_try_with_resource_with_useless_assertAll() {
+    try(org.assertj.core.api.AutoCloseableSoftAssertions softly = new org.assertj.core.api.AutoCloseableSoftAssertions()) {
+      softly.assertThat(1).isLessThan(2);
+      softly.assertAll();
+    } // Noncompliant {{Add one or more 'assertThat' before 'assertAll'.}}
```

Comment:
For this case, the message should be modified in order to be more explicit (the `assertAll` which will be called is implicit):
`Add one or more 'assertThat' before the end of the 'try-with-resource' block or remove the last 'assertAll'.`